content/en/docs/tasks/job/job-with-pod-to-pod-communication.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Job with Pod-to-Pod Communication
+content_type: task
+min-kubernetes-server-version: v1.21
+weight: 30
+---
+
+<!-- overview -->
+
+In this example, you will run a Job in [Indexed completion mode](https://kubernetes.io/blog/2021/04/19/introducing-indexed-jobs/) configured such that
+the pods created by the Job can communicate with each other using pod hostnames rather than pod IP addresses.
+
+Pods within a Job might need to communicate among themselves. The user workload running in each pod could query the Kubernetes API server
+to learn the IPs of the other Pods, but it's much simpler to rely on Kubernetes' built-in DNS resolution.
+
+Jobs in Indexed completion mode automatically set the pods' hostname to be in the format of
+`${jobName}-${completionIndex}`. You can use this format to deterministically build
+pod hostnames and enable pod communication *without* needing to create a client connection to
+the Kubernetes control plane to obtain pod hostnames/IPs via API requests. 
+
+This configuration is useful
+for use cases where pod networking is required but you don't want to depend on a network 
+connection with the Kubernetes API server.
+
+## {{% heading "prerequisites" %}}
+
+You should already be familiar with the basic use of [Job](/docs/concepts/workloads/controllers/job/).
+
+{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}
+
+{{<note>}} If you are using MiniKube or a similar tool, you may need to take [extra steps](https://minikube.sigs.k8s.io/docs/handbook/addons/ingress-dns/) to ensure you have DNS. {{</note>}}
+
+<!-- steps -->
+
+## Starting a Job with Pod-to-Pod Communication
+
+To enable pod-to-pod communication using pod hostnames in a Job, you must do the following:
+
+1. Set up a [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services)
+with a valid label selector for the pods created by your Job. The headless service must be in the same namespace as 
+the Job. One easy way to do this is to use the `job-name: <your-job-name>` selector, since the `job-name` label will be automatically added by Kubernetes. This configuration will trigger the DNS system to create records of the hostnames of 
```

Comment:
```suggestion
   One easy way to do this is to use the `job-name: <your-job-name>` selector, since the `job-name`
   label will be automatically added by Kubernetes. This configuration will trigger the DNS system to
   create records of the hostnames of the pods running your Job.
```